lib/beacon_web/controllers/error_html.ex

```diff
@@ -1,21 +1,21 @@
 defmodule BeaconWeb.ErrorHTML do
   @moduledoc false
-
   use BeaconWeb, :html
 
-  # If you want to customize your error pages,
-  # uncomment the embed_templates/1 call below
-  # and add pages to the error directory:
-  #
-  #   * lib/beacon_web/controllers/error/404.html.heex
-  #   * lib/beacon_web/controllers/error/500.html.heex
-  #
-  # embed_templates "error/*"
+  def render(template, assigns) do
+    {_, _, %{extra: %{session: %{"beacon_site" => site}}}} = assigns.conn.private.phoenix_live_view
+
+    status =
+      template
+      |> String.split(".")
+      |> hd()
+      |> String.to_integer()
+
+    %{layout: %{template: layout_template}, template: page_template} =
+      site
+      |> Beacon.Content.get_error_page(status)
+      |> Beacon.Repo.preload(:layout)
 
-  # The default is to render a plain text page based on
-  # the template name. For example, "404.html" becomes
-  # "Not Found".
-  def render(template, _assigns) do
-    Phoenix.Controller.status_message_from_template(template)
+    EEx.eval_string(layout_template, assigns: [inner_content: page_template])
   end
```

Comment:
I think we need to be more strict to avoid errors here, something like:

```elixir
  def render(
        <<status_code::binary-size(3), _rest::binary>>,
        %{
          conn: %{private: %{phoenix_live_view: {_, _, %{extra: %{session: %{"beacon_site" => site}}}}}}
        }
      ) do
    status_code
    |> String.to_integer()
    |> render_error_page(site)
  rescue
    _ -> render_default_error_page()
  end

  defp render_error_page(status, site) when status >= 400 and status <= 599 and is_atom(site) do
    error_page =
      site
      |> Beacon.Content.get_error_page(status)
      |> Beacon.Repo.preload(:layout)

    case error_page do
      %{layout: %{template: layout_template}, template: page_template} ->
        EEx.eval_string(layout_template, assigns: [inner_content: page_template])

      nil ->
        "render some default error page for the given status code"
    end
  end

  defp render_default_error_page do
    "render a default error page here"
  end
```

Reply:
This would all be replaced with the module recompilation approach we discussed last week, right?

Reply:
Yes part of it would be replaced. If you intended to work on it then it's fine to keep the existing code since it will be replaced soon.